java-client/src/test/java/org/opensearch/client/opensearch/experiments/ParsingTests.java

```diff
@@ -88,11 +80,40 @@ public void testFoo() throws Exception {
   }
 
   @Test
-  public void testIndexSettingsMappingParsing() {
+  public void testIndexSettingsTranslogOptionsParsing() {
 
     try {
 
-      var baos = new ByteArrayOutputStream();
+      var indexSettings = IndexSettings.of(_1 -> _1.translog(Translog.of(tr -> tr
+              .syncInterval(Time.of(t -> t.time("10s")))
+              .durability("async")
+              .flushThresholdSize("256mb"))));
+
+      var str = serialize(indexSettings);
+      assertEquals("{\"translog\":{\"durability\":\"async\",\"flush_threshold_size\":\"256mb\"," +
+              "\"sync_interval\":\"10s\"}}", str);
+
+      IndexSettings deserialized = deserialize(str, IndexSettings._DESERIALIZER);
+      assertEquals(indexSettings.translog().syncInterval().time(), deserialized.translog().syncInterval().time());
+      assertEquals(indexSettings.translog().durability(), deserialized.translog().durability());
+      assertEquals(indexSettings.translog().flushThresholdSize(), deserialized.translog().flushThresholdSize());
+
+      var deprecatedForm = "{\"translog\":{\"sync_interval\":\"10s\"},\"translog.durability\":\"async\",\"translog" +
+              ".flush_threshold_size\":\"256mb\"}";
+      IndexSettings deprecatedDeserialized = deserialize(deprecatedForm, IndexSettings._DESERIALIZER);
+      assertEquals(indexSettings.translog().syncInterval().time(), deprecatedDeserialized.translog().syncInterval().time());
+      assertEquals(indexSettings.translog().durability(), deprecatedDeserialized.translog().durability());
+      assertEquals(indexSettings.translog().flushThresholdSize(), deprecatedDeserialized.translog().flushThresholdSize());
+
+    } catch (JsonParsingException je) {
```

Comment:
What's the difference between the caught `JsonParsingException` and the re-thrown one? Either way this is a test and it should never hit the `catch` part, so I suggest removing it.

Reply:
Agreed, fixed and force-pushed.